src/callbacks_step/summary.jl

```diff
@@ -14,42 +22,160 @@ function Base.show(io::IO, cb::DiscreteCallback{Condition,Affect!}) where {Condi
 end
 
 
+# Format a key/value pair for output from the SummaryCallback
+function format_key_value_line(key::AbstractString, value::AbstractString, key_width, total_width;
+                               indentation_level=0, guide='…', filler='…', prefix="│ ", suffix=" │")
+  @assert key_width < total_width
+  line  = prefix
+  if indentation_level == 0
+    squeezed_key = squeeze(key, key_width, filler=filler)
+  else
+    indentation = prefix^indentation_level
+    reduced_key_width = key_width - length(indentation)
+    squeezed_key = indentation * squeeze(key, reduced_key_width, filler=filler)
+  end
+  line *= squeezed_key
+  line *= ": "
+  short = key_width - length(squeezed_key)
+  if short <= 1
+    line *= " "
+  else
+    line *= guide^(short-1) * " "
+  end
+  value_width = total_width - length(prefix) - length(suffix) - key_width - 2
+  squeezed_value = squeeze(value, value_width, filler=filler)
+  line *= squeezed_value
+  short = value_width - length(squeezed_value)
+  line *= " "^short
+  line *= suffix
+
+  @assert length(line) == total_width "should not happen: algorithm error!"
+
+  return line
+end
+format_key_value_line(key, value, args...; kwargs...) = format_key_value_line(string(key), string(value), args...; kwargs...)
+
+# Squeeze a string to fit into a maximum width by deleting characters from the center
+function squeeze(message, max_width; filler='…')
```

Comment:
```suggestion
function squeeze(message, max_width; filler::Char='…')
```
You assume `length(filler) == 1` in the code below, so it might make sense to impose this restriction also in the calling structure.

Reply:
Good catch, thanks.